lsp-mode.el

```diff
@@ -3091,7 +3091,9 @@ active `major-mode', or for all major modes when ALL-MODES is t."
   (folder->servers (make-hash-table :test 'equal))
   ;; ‘metadata’ is a generic storage for workspace specific data. It is
   ;; accessed via `lsp-workspace-set-metadata' and `lsp-workspace-set-metadata'
-  (metadata (make-hash-table :test 'equal)))
+  (metadata (make-hash-table :test 'equal))
+  ;; If we should warn user about a possible inconsistency between LSP packages.
+  (warn-possible-inconsistency? lsp-use-plists))
```

Comment:
Adding this key will break session deserialization if you have serialized the old one without this field(not sure if the issue was present in all emacs versions).

Also, take a look at lsp--persist-session  - it does not serialize all keys. But lsp-session-metata might contain stuff that is not serializable like markers. So I guess we should serialize only certain keys from the metadata.

Reply:
@yyoncho Done, moved the new boolean to metadata and on `lsp--persist-session`, save only this new field of metadata.